pkg/webui/console/containers/gateway-managed-gateway/connection-settings/index.js

```diff
@@ -219,21 +224,33 @@ const GatewayConnectionSettings = () => {
         const { entities } = await dispatch(attachPromise(getCollaboratorsList('gateway', gtwId)))
         collaborators = entities
       }
-      const { wifiProfile, entityId } = await fetchWifiProfile(collaborators)
-      const { ethernetProfile } = await fetchEthernetProfile()
-      const isNonSharedProfile = Boolean(wifiProfile) && !Boolean(wifiProfile.data.shared)
-      if (isNonSharedProfile) {
-        setNonSharedWifiProfileId(selectedManagedGateway.wifi_profile_id)
+      let newValues = {}
+      if (hasWifi) {
```

Comment:
You are missing `hasWifi`, `hasEthernet` and `initialValues` in the dependency array of the function. Since we don't want `initialValues` in the dependency array because of infinite loops, I suggest refactoring this function to:

```
const loadData = useCallback(
    async dispatch => {
      const fetchCollaborators = async () => {
        if (mayViewCollaborators) {
          const { entities } = await dispatch(attachPromise(getCollaboratorsList('gateway', gtwId)))
          return entities
        }
        return []
      }

      const collaborators = await fetchCollaborators()
      const newValues = {}

      if (hasWifi) {
        const { wifiProfile, wifiProfileEntityId } = await fetchWifiProfile(collaborators)
        const isNonSharedWifiProfile = Boolean(wifiProfile) && !Boolean(wifiProfile.data.shared)

        if (isNonSharedWifiProfile) {
          setNonSharedWifiProfileId(selectedManagedGateway.wifi_profile_id)
        }

        newValues.wifi_profile = updateInitialWifiProfile(
          wifiProfile,
          wifiProfileEntityId,
          isNonSharedWifiProfile,
        )
      }

      if (hasEthernet) {
        const { ethernetProfile } = await fetchEthernetProfile()
        newValues.ethernet_profile = updateInitialEthernetProfile(ethernetProfile)
      }

      setInitialValues(oldValues => ({ ...oldValues, ...newValues }))
    },
    [
      fetchEthernetProfile,
      fetchWifiProfile,
      gtwId,
      hasEthernet,
      hasWifi,
      mayViewCollaborators,
      selectedManagedGateway.wifi_profile_id,
      updateInitialEthernetProfile,
      updateInitialWifiProfile,
    ],
  )
```

Also please change `updateInitialWifiProfile` and `updateInitialEthernetProfile` functions with:

```
const updateInitialWifiProfile = useCallback(
    (profile, entityId, isNonSharedProfile) => ({
      ...(isNonSharedProfile && { ...profile.data }),
      profile_id: isNonSharedProfile
        ? 'non-shared'
        : (selectedManagedGateway.wifi_profile_id ?? ''),
      _override: !Boolean(profile) && hasWifiProfileSet,
      _enable_wifi_connection: Boolean(profile),
      _profile_of: entityId ?? '',
    }),
    [hasWifiProfileSet, selectedManagedGateway.wifi_profile_id],
  )

  const updateInitialEthernetProfile = useCallback(
    profile => ({
      ...revertEthernetProfile(profile?.data ?? {}),
      profile_id: selectedManagedGateway.ethernet_profile_id ?? '',
    }),
    [selectedManagedGateway.ethernet_profile_id],
  )
```